### PR TITLE
Modifying test_target_update_if.c to use headerfile (Issue #10)

### PR DESCRIPTION
**Pull request by tmh97**
Test is ready for review, passes with clang and gcc on the device. I noticed that test fails on the host when NO_OFFLOADING is enabled for GCC compilation, however, clang passes fine on the host even with NO_OFFLOADING. Need to decide if this is a concern or not considering it is a 'target' test.

### CODE REVIEW

File: tests/4.5/target_update/test_target_update_if.c [hunk 1]
```diff
@@ -39,29 +41,26 @@ int init_b(){
...
   // a and b array initialization
   for (i = 0; i < N; i++) {
     b[i] = 2; 
     c[i] = 0;
   }
-#pragma omp target data map(to: a[:N], b[:N]) map(tofrom: c) map(tofrom: isHost)
+#pragma omp target data map(to: a[:N], b[:N]) map(tofrom: c)
 {
-  #pragma omp target //map(to: a[:N], b[:N], isHost)
+  #pragma omp target //map(to: a[:N], b[:N])
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: remove comment

File: tests/4.5/target_update/test_target_update_if.c [hunk 2]
```diff
@@ -39,29 +41,26 @@ int init_b(){
...
 
   // We test for offloading
-#pragma omp target map(from: isOffloading)
-  {
-    isOffloading = !omp_is_initial_device();
-  }
+  int is_offloading;
+  OMPVV_TEST_AND_SET_OFFLOADING(is_offloading); 
 
 for(count=0; count< 4;count++){
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: spaces between `<`,  `=`  and  `;`

File: tests/4.5/target_update/test_target_update_if.c [hunk 3]
```diff
@@ -39,29 +41,26 @@ int init_b(){
...
   // a and b array initialization
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: b and c are initialized

File: tests/4.5/target_update/test_target_update_if.c [hunk 4]
```diff
@@ -71,7 +70,7 @@ for(count=0; count< 4;count++){
   change_flag = init_b();
   #pragma omp target update if (change_flag) to(b[:N]) //update b=4 for all odd iterations
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: Is this comment of b=4 accurate?

File: tests/4.5/target_update/test_target_update_if.c [hunk 5]
```diff
@@ -71,7 +70,7 @@ for(count=0; count< 4;count++){
...
 
-  #pragma omp target //map(from: a,b,c) map(tofrom: isHost)//mapping from to counter default tofrom mapping on b
+  #pragma omp target //map(from: a,b,c) //mapping from to counter default tofrom mapping on b
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: remove the map(from) and potentially the comment. It's ok if you have a comment here explaining what's the purpose, but we need to fix this wording I believe.

File: tests/4.5/target_update/test_target_update_if.c [hunk 6]
```diff
@@ -71,7 +70,7 @@ for(count=0; count< 4;count++){
...
   {
         int j = 0;
         for (j = 0; j < N; j++) {
```
Comment (josemonsalve2) [inline tests/4.5/target_update/test_target_update_if.c]: Line 83: Spaces of comment

File: tests/4.5/target_update/test_target_update_if.c (hunk 4 above)
Comment (tmh97) [reply tests/4.5/target_update/test_target_update_if.c]: Good catch, I do not believe this is accurate. Will look at this closer and fix.

Comment (tmh97) [reply tests/4.5/target_update/test_target_update_if.c]: @josemonsalve2 looked into this, b=4 is correct value during "odd" iterations. This is good to go.

**Pull request merged**